inmem/session_store.go

```diff
@@ -0,0 +1,82 @@
+package inmem
+
+import (
+	"sync"
+	"time"
+)
+
+type SessionStore struct {
+	data map[string]string
+
+	timers map[string]*time.Timer
+
+	mu sync.RWMutex
+}
+
+func NewSessionStore() *SessionStore {
+	return &SessionStore{
+		data:   map[string]string{},
+		timers: map[string]*time.Timer{},
+	}
+}
+
+func (s *SessionStore) Set(key, val string, expireAt time.Time) error {
+	if !expireAt.IsZero() && expireAt.Before(time.Now()) {
+		// key is already expired. no problem
+		return nil
+	}
+
+	s.mu.Lock()
+	s.data[key] = val
+	s.mu.Unlock()
+
+	if !expireAt.IsZero() {
+		return s.ExpireAt(key, expireAt)
```

Comment:
Take it or leave it nit: You could do this in one lock; if you had an un-exported `expireAt` method which both this and `ExpireAt` exported both used.